src/main/java/com/atlassian/db/replica/internal/aurora/RdsDns.java

```diff
@@ -0,0 +1,83 @@
+package com.atlassian.db.replica.internal.aurora;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class RdsDns {
+    private static final Pattern DNS_PATTERN = Pattern.compile("([^.]+).([^:]*):?([0-9]+)?");
+    private final String region;
+    private final String domain;
+    private final Integer port;
+
+    public RdsDns(String region, String domain, Integer port) {
```

Comment:
Which ones are nullable? AFAIK all API so far was non-null. This means we need to start adding `@Nullable` and `@NonNull`.

Reply:
Port can be nullable